.github/workflows/js.yml

```diff
@@ -81,9 +81,9 @@ jobs:
         run: archery docker push debian-js
 
   macos:
-    name: AMD64 macOS 11 NodeJS ${{ matrix.node }}
-    runs-on: macos-latest
-    if: github.event_name == 'push'
+    name: AMD64 macOS 12 NodeJS ${{ matrix.node }}
+    runs-on: macos-12
+    if: ${{ !contains(github.event.pull_request.title, 'WIP') }}
     timeout-minutes: 90
```

Comment:
Could you reduce timeout to 30 or something?

https://github.com/apache/arrow/actions/runs/9550814670/job/26323703330?pr=42175 took 18m 51s.

Reply:
Fixed it!